### PR TITLE
Clean up deleted Elasticsearch annotations in Postgres

### PR DESCRIPTION
**Pull request by chdorner**
After each run of the migration script, identify all the annotations
that were previously imported but have been deleted in Elasticsearch and
delete them from Postgres as well.

### CODE REVIEW

File: scripts/migrate.py [hunk 1]
```diff
@@ -283,5 +313,22 @@ def _batch_iter(n, iterable):
...
+    annotations = scan(es)
+    for batch in _batch_iter(BATCH_SIZE, annotations):
+        for ann in batch:
+            ids.add(ann['_id'])
+
+    return ids
+
+
+def collect_pg_annotations():
+   ids = set()
```
Comment (nickstenning) [inline scripts/migrate.py]: Seeing as this returns annotations, not IDs, I suspect you might as well call `Session.query(Annotation).all()` above and do away with this function.

Comment (chdorner) [inline scripts/migrate.py]: Does `.all()` use the Postgres streaming functionality automatically behind the scenes?

**Comment by nickstenning** [comment]
Well, that's one way of doing it...!

Loading all 300K annotations into memory does seem like a pretty brutal approach to take here. Perhaps something like this could be a slightly less blunt approach:

```python

in_postgres = set((a.id for a in Session.query(Annotation.id)))

for batch in _batch_iter(BATCH_SIZE, scan(es)):
    in_postgres.difference_update((a['_id'] for a in batch))

Session.query(Annotation).filter(Annotation.id.in_(in_postgres)).remove()
Session.commit()
```

File: scripts/migrate.py (hunk 1 above)
Comment (nickstenning) [inline scripts/migrate.py]: You're collecting them all in one big set anyway, so I `yield_per` doesn't buy you anything. I suspect you previously had this function returning a set of ids, no?

**Comment by nickstenning** [comment]
LGTM.

**Pull request merged**